src/streamer/jamstreamer.cpp

```diff
@@ -0,0 +1,37 @@
+#ifndef _WIN32
+#include "jamstreamer.h"
+
+using namespace streamer;
+
+// --- CONSTRUCTOR ---
+CJamStreamer::CJamStreamer() {}
+
+// --- PROCESS ---
+// put the received pcm data into the pipe to ffmpeg
+void CJamStreamer::process( int iServerFrameSizeSamples, CVector<int16_t> data ) {
+    if (pcm == 0) // thanks to hselasky
+    {
+        pcm = new int16_t [2 * iServerFrameSizeSamples];
+    }
+    for ( int i = 0; i < ( 2 * iServerFrameSizeSamples ); i++ )
+    {
+        pcm[i] = data[i];
+    }
+    fwrite(pcm, 2, ( 2 * iServerFrameSizeSamples ), pipeout);
```

Comment:
The copy to `pcm` is redundant since vector contents are guaranteed to be contiguous. So you can put `&data[0]` instead of `pcm` here and skip the copy loop.

Reply:
thanks for the advice! did it and it worked :)